src/main/java/org/gridsuite/modification/server/modifications/VoltageInitModification.java

```diff
@@ -4,264 +4,304 @@
-                    return;
+                    reporter.report(Report.builder().withKey("2WindingsTransformerNotFound")
+                                                    .withDefaultMessage("2 windings transformer with id=${id} not found")
+                                                    .withValue("id", t.getTransformerId())
+                                                    .withSeverity(TypedValue.WARN_SEVERITY).build());
+                } else if (twoWindingsTransformer.getRatioTapChanger() == null) {
+                    reporter.report(Report.builder().withKey("2WindingsTransformerRatioTapChangerNotFound")
+                                                    .withDefaultMessage("2 windings transformer with id=${id} : Ratio tap changer not found")
+                                                    .withValue("id", t.getTransformerId())
+                                                    .withSeverity(TypedValue.WARN_SEVERITY).build());
+                } else {
+                    reporter.report(Report.builder().withKey("2WindingsTransformerModification")
+                                                    .withDefaultMessage("2 windings transformer with id=${id} modified :")
+                                                    .withValue("id", t.getTransformerId())
+                                                    .withSeverity(TypedValue.TRACE_SEVERITY).build());
+                    final int oldTapPosition = twoWindingsTransformer.getRatioTapChanger().getTapPosition();
+                    twoWindingsTransformer.getRatioTapChanger().setTapPosition(t.getRatioTapChangerPosition());
+                    reporter.report(ModificationUtils.buildModificationReport(oldTapPosition, t.getRatioTapChangerPosition(), "Ratio tap changer position", 1, TypedValue.TRACE_SEVERITY));
                 }
-
-                Reporter reporter = subReporter.createSubReporter(TWO_WINDINGS_TRANSFORMER_MSG + t.getTransformerId(), TWO_WINDINGS_TRANSFORMER_MSG + t.getTransformerId());
-                report(reporter, "2WindingsTransformerModification", "2 windings transformer with id=${id} modified :", Map.of("id", t.getTransformerId()), TypedValue.INFO_SEVERITY);
-
-                int oldTapPosition = twoWindingsTransformer.getRatioTapChanger().getTapPosition();
-                twoWindingsTransformer.getRatioTapChanger().setTapPosition(t.getRatioTapChangerPosition());
-                reporter.report(ModificationUtils.getInstance().buildModificationReportWithIndentation(oldTapPosition, t.getRatioTapChangerPosition(), "Ratio tap changer position", 1));
             }
-        });
+        }
+        if (modificationsCount > 0) {
+            subReporter.report(new Report("windingsTransformerModificationsResume", "${count} transformer(s) have been modified.", Map.of(
```

Comment:
```suggestion
            String message = modificationsCount == 1 ? "${count} transformer has been modified." : "${count} transformers have been modified.";
            subReporter.report(new Report("windingsTransformerModificationsResume", message, Map.of(
```

Reply:
same

Reply:
For clarity purpose, I think using ternary operator like this is not the best